lib/BaseLogger.js

```diff
@@ -56,21 +56,27 @@ class BaseLogger {
   }
 
   _writeLog(level, message) {
-    // We may have sent an Error;
-    // if we have, the stack is most useful.
-    if (message && message.stack) {
-      return this._writeLog(level, message.stack);
-    }
-
     const logData = {
       timestamp: moment.utc(),
       source: this.source,
       level,
       message
     };
+    // We may have sent an Error;
+    // if we have, the stack is most useful
+    if (message && message.stack) {
+      logData.stack = message.stack;
+    }
+    // Errors tend to already have a message
+    if (message && message.message) {
+      logData.message = message.message;
+    }
+    // If we are logging an error, key may be the type
+    if (message && message.key) {
+      logData.key = message.key;
+    }
```

Comment:
Instead of all of this, how about we just augment `message` with `timestamp`, `source` and `level`?